public/hooks/use_get_chunks_from_http_response.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BehaviorSubject } from 'rxjs';
+import { LLMResponseType, useChatState } from './use_chat_state';
+import { streamDeserializer } from '../../common/utils/stream/serializer';
+import { StreamChunk } from '../../common/types/chat_saved_object_attributes';
+import { MessageContentPuller } from '../utils/message_content_puller';
+import { convertEventStreamToObservable } from '../../common/utils/stream/stream_to_observable';
+
+export const useGetChunksFromHTTPResponse = () => {
+  const { chatStateDispatch } = useChatState();
+
+  const getConsumedChunk$FromHttpResponse = async (props: {
+    stream: ReadableStream;
+    abortController: AbortController;
+  }) => {
+    const chunk$ = new BehaviorSubject<StreamChunk | undefined>(undefined);
+    const messageContentPuller = new MessageContentPuller();
+    const result = convertEventStreamToObservable(props.stream, props.abortController);
+    props.abortController.signal.addEventListener('abort', () => {
+      messageContentPuller.stop();
+      result.cancel();
+    });
+    chatStateDispatch({
+      type: 'updateResponseType',
+      payload: {
+        type: LLMResponseType.STREAMING,
+      },
+    });
+    result.output$.subscribe({
+      next: (chunk: string) => {
+        try {
+          const chunkObjects = streamDeserializer(chunk);
+          for (const chunkObject of chunkObjects) {
+            if (chunkObject.event === 'appendMessageContent') {
+              messageContentPuller.addMessageContent(
+                chunkObject.data.messageId,
+                chunkObject.data.content
+              );
+            } else {
+              chunk$.next(chunkObject);
+            }
+          }
+        } catch (e) {
+          chunk$.error(e);
```

Comment:
Nit: Do we need to call `messageContentPuller.inputComplete();` or `messageContentPuller.stop();` here?

Reply:
Sure, updated.